src/components/sw360/FossologyClearing/FossologyClearing.tsx

```diff
@@ -24,358 +25,394 @@ import { FossologyProcessInfo, FossologyProcessStatus } from '@/object-types/Fos
+        show: false,
+    })
+
+    const [progressStatus, setProgressStatus] = useState({
+        percent: 0,
+        stepName: '',
+    })
+
+    const resetTimeCountDown = () => {
+        setTimeInterval(5)
+    }
+
+    const clearAllInterval = useCallback(() => {
+        countDownInterval.current = clearInterval(countDownInterval.current)
+        progressInterval.current = clearInterval(progressInterval.current)
+        resetTimeCountDown()
+    }, [])
+
+    const fetchData = useCallback(
+        async (url: string) => {
+            return ApiUtils.GET(url, session.user.access_token)
+                .then((response) => response.json())
+                .catch(() => undefined)
+        },
+        [session]
+    )
+
+    const fetchRelease = useCallback(async () => {
+        const url = `releases/${releaseId}`
+        fetchData(url).then((data) => {
+            setRelease(data)
+            numberOfSourceAttachment.current = countSourceAttachment(data._embedded['sw360:attachments'])
+            if (!isClearingAllowed()) {
+                clearAllInterval()
+                showMessage(clearingMessages.NUMBER_OF_ATTACHMENTS_NOT_MATCH)
+                setProgressStatus({
+                    percent: 0,
+                    stepName: '',
+                })
+                return
+            }
+        })
+    }, [releaseId, clearAllInterval, fetchData])
+
+    const handleCloseDialog = () => {
+        hideMessage()
+        clearAllInterval()
+        setProgressStatus({
+            percent: 0,
+            stepName: '',
+        })
+        setRelease(undefined)
+        setShow(false)
+    }
+
+    const handleOutdated = () => {
+        handleFossologyClearing({ markFossologyProcessOutdated: true })
+        showMessage(clearingMessages.SET_OUTDATED)
+        setConfirmShow(false)
+    }
+
+    const triggerFossologyClearing = useCallback(
+        async (params: { [key: string]: string }) => {
+            const url = CommonUtils.createUrlWithParams(`releases/${releaseId}/triggerFossologyProcess`, params)
+            const response = await fetchData(url)
+            return response ? true : false
+        },
+        [releaseId, fetchData]
+    )
+
+    const checkFossologyProcessStatus = useCallback(async () => {
+        const url = `releases/${releaseId}/checkFossologyProcessStatus`
+        const response: FossologyProcessStatus = await fetchData(url)
+
+        if (response.status === 'SUCCESS') {
+            setProgressStatus({
+                percent: PERCENT_DONE,
+                stepName: 'Report generation done',
+            })
+            showMessage(clearingMessages.CLEARING_SUCCESS)
+            clearAllInterval()
+            return
+        }
+
+        if (response.fossologyProcessInfo !== null) {
+            updateProgressStatus(response.fossologyProcessInfo)
+        }
+        // RECURSIVE CALL
+        // if (progressStatus.percent < PERCENT_DONE && progressStatus.percent > 0) {
```

Comment:
We need this code snippet to run the process after closing and reopening the modal.  But i found a small bug with this. I will fix it later

Reply:
Done